lib/fika/compiler/parser/function_def.ex

```diff
@@ -63,12 +79,49 @@ defmodule Fika.Compiler.Parser.FunctionDef do
     |> wrap(exps)
     |> concat(require_space)
     |> ignore(string("end"))
-    |> label("function definition")
-    |> Helper.to_ast(:function_def)
+    |> label("public function definition")
+    |> Helper.to_ast(:public_function_def)
 
-  function_defs =
+  ext_atom =
+    ignore(string("\""))
+    |> repeat(choice([string("\\\""), utf8_char(not: ?")]))
+    |> ignore(string("\""))
+    |> Helper.to_ast(:ext_atom)
+
+  ext_function_def =
     allow_space
-    |> concat(function_def)
+    |> ignore(string("ext"))
+    |> concat(require_space)
+    |> concat(identifier)
+    |> concat(arg_parens)
+    |> concat(return_type)
+    |> concat(allow_space)
+    |> ignore(string("="))
```

Comment:
Finally, is there a reason to using `=` here?
I'd rather use `do` or `->` which are more "clause-related", keeping `=` for a comparison/matching context only

Reply:
Had the same concern when writing this! I am not so happy about the `=` here either. Good thing is that it's easy to change and we'll do it when we revisit our function def syntax and possibly introduce the `->` symbol and make it consistent.